lms/migrations/versions/f3d631c110bf_canvas_files_enabled.py

```diff
@@ -0,0 +1,81 @@
+"""
+Make canvas files enabled an AI setting.
+
+Revision ID: f3d631c110bf
+Revises: 0b7391f5b1e3
+Create Date: 2022-12-28 10:15:40.830444
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+revision = "f3d631c110bf"
+down_revision = "5f0057d6d60f"
+
+
+def upgrade():
+    """
+    Explicitly store the `files_enabled` setting for canvas instances.
+
+    Currently the logic for this "setting" is:
+
+    ```
+    enabled = (request.product.family == Product.Family.CANVAS) and (
+        "custom_canvas_course_id" in request.lti_params
+        and application_instance.developer_key is not None
+    )
+    ```
+
+    Querying for just:
+
+        select count(id) from application_instances where developer_key is not null
+
+    returns 957 AIs at the time of writing
+
+    while trying to get the 'custom_canvas_course_id' part:
+
+        select
+            count(id)
+        from application_instances
+        where
+            developer_key is not null
+            and exists (
+                select
+                    id
+                from grouping
+                where
+                    type ='course'
+                    and grouping.application_instance_id = application_instances.id
+                    and extra->'canvas'->>'custom_canvas_course_id' is not null
+            )
+
+    we get 491 AIs
+
+    but removing the last condition, the one with custom_canvas_course_id gives us the same number.
+
+    This means that all canvas instances that have a developer key and have created courses do send custom_canvas_course_id.
```

Comment:
I reckon this means that the only risk here is leaving this disabled for schools that:

- do have a developer_key
- have the configuration to send custom_canvas_course_id in place
- haven't ever launched a course
- decide to start using it now

But I'd like have other pair of eyes have a look.

Reply:
Alternatively we could just enable if for all the AI that have the developer key, see:

https://github.com/hypothesis/lms/blob/18c0648b3edbdf3ade18ff405500bfb89b55e846/lms/migrations/versions/da66efab9e5f_enable_canvas_groups.py